libs/parallelism/algorithms/include/hpx/parallel/algorithms/transform.hpp

```diff
@@ -530,32 +508,89 @@ namespace hpx { namespace parallel { inline namespace v1 {
             {
             }
 #endif
-
             transform_binary_iteration& operator=(
                 transform_binary_iteration const&) = default;
             transform_binary_iteration& operator=(
                 transform_binary_iteration&&) = default;
 
-            template <typename Iter>
+            template <typename Iter, typename F_ = fun_type>
             HPX_HOST_DEVICE HPX_FORCEINLINE
                 hpx::tuple<typename hpx::tuple_element<0,
                                typename Iter::iterator_tuple_type>::type,
                     typename hpx::tuple_element<1,
                         typename Iter::iterator_tuple_type>::type,
                     typename hpx::tuple_element<2,
                         typename Iter::iterator_tuple_type>::type>
-                operator()(Iter part_begin, std::size_t part_size,
-                    std::size_t /*part_index*/)
+                operator()(Iter part_begin, std::size_t part_size, std::size_t)
             {
+                hpx::util::annotate_function _(f_);
```

Comment:
Is this supposed to stay here?

Reply:
It was there before... I made it consistent, only.

Reply:
Hrm, indeed. Now the question is, is this going to overwrite the annotation set by the policy? I think it will. For the sequential cases it makes sense to set the annotation explicitly, but for the parallel cases I'm not so sure.

Reply:
You're right. The sequential cases of all algorithms ought to be changed to go through the executor anyways (can be done in one place), so I'd say let's remove the annotation there as well.

Reply:
I have removed the annotations here and from for_each now.